### PR TITLE
Issue-4735 JSF deployment failure due to UnsupportedOperationExceptio…

### PR DESCRIPTION
**Pull request by RanabirChakraborty**
Signed-off-by: Ranabir Chakraborty rchakrab@redhat.com

Issue: [#4735 ](https://github.com/eclipse-ee4j/mojarra/issues/4735)

Wildfly issue: https://issues.redhat.com/browse/WFLY-13792

2.3 branch PR: https://github.com/eclipse-ee4j/mojarra/pull/4825

This problem happens always when the JavaEE server is servlet 4.0 complaint as [Undertow](https://undertow.io/)

### CODE REVIEW

**Comment by fjuma** [comment]
Just FYI, this PR has been submitted against master but we'll also need this to be merged to the 2.3 branch.

**Comment by RanabirChakraborty** [comment]
Opened a PR for 2.3 https://github.com/eclipse-ee4j/mojarra/pull/4825

**Review by manorrock — APPROVED** [review]
Needs approval from another committer as no tests were supplied

**Pull request merged**
